src/datapool/datapool_pmem.c

```diff
@@ -153,14 +163,22 @@ datapool_flag_clear(struct datapool *pool, int flag)
  * finish successfully.
  */
 struct datapool *
-datapool_open(const char *path, size_t size, int *fresh, bool prefault)
+datapool_open(const char *path, const char* user_signature, size_t size, int *fresh, bool prefault)
 {
     struct datapool *pool = cc_alloc(sizeof(*pool));
     if (pool == NULL) {
         log_error("unable to create allocate memory for pmem mapping");
         goto err_alloc;
     }
 
+    /* user signature cannot be empty and length must be
+     * appropriate to fit in pool->hdr->user_signature */
+    if (user_signature == NULL ||
+            (cc_strnlen(user_signature, DATAPOOL_USER_LAYOUT_LEN) == DATAPOOL_USER_LAYOUT_LEN )) {
+        log_error("wrong user signature");
```

Comment:
I meant in the log message ;P

Reply:
Done - log_error was extended with additional information here and after datapool_valid_user_signature method